core/src/main/java/org/pentaho/di/core/database/Database.java

```diff
@@ -1962,7 +1962,8 @@ private ResultSet getTableMetaData( String schema, String table ) throws KettleD
       throw new KettleDatabaseException( BaseMessages.getString( PKG, "Database.Error.UnableToGetDbMeta" ) );
     }
     try {
-      tables = getDatabaseMetaData().getTables( null, schema, table, TABLE_TYPES_TO_GET );
+      tables = databaseMeta.getDatabaseInterface().getTables(
```

Comment:
`databaseMeta.getDatabaseInterface().getTables(getDatabaseMetaData(), databaseMeta, ...` is quite a mouthful.

I wonder- would it make sense to add a .getTables method to DatabaseMeta that is a pass through to the databaseinterface?  That way the call could be simplified to something like:

 `databaseMeta.getTables(getDatabaseMetaData(),schema, table, TABLE_TYPES_TO_GET );`

Even that's a bit confusing with "databaseMeta" and "getDatabaseMetaData", but at least visually scans a bit better.

Reply:
Maybe even bring a .getTables() up in to Database, delegating to the dbinterface.  Pretty easy to grok `Database.getTables(schema, table, types)`